analysis/mixpanel/mixpanel_retention.sql

```diff
@@ -0,0 +1,84 @@
+with events as 
+(
+	-- can change 'week' to 'day', 'month', 'qtr', etc.
+	-- Make sure to make corresponding everywhere below
+	select date_trunc('week', ac_timestamp)::date as ac_date, *
+	from {{env.schema}}.mixpanel_export
+),
+
+users as
+(
+	select ac_user_id, min(date_trunc('week', ac_user_created)::date) as ac_user_created
+	from {{env.schema}}.mixpanel_engage
+	group by ac_user_id
+),
+
+overall_retention as
+(
+	select *, 100.0*retained_users/active_users as overall_retention
+	from
+	(
+		select
+			events.ac_date,
+			count(distinct events.ac_user_id) as active_users,
+			count(distinct future_events.ac_user_id) as retained_users
+		from events
+		-- join future events 
+		left outer join events as future_events
+			on events.ac_user_id = future_events.ac_user_id
+			and events.ac_date = future_events.ac_date - interval '1 week'
+		group by events.ac_date
+	)
+),
+
+new_retention as
+(
+	select *, 100.0*retained_users/active_users as new_retention
```

Comment:
why multiply by 100? better to leave between 1 and zero so end tool can format it...